codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGenerator.kt

```diff
@@ -189,6 +189,11 @@ class BuilderGenerator(
         metadata.derives.filter {
             it == RuntimeType.Debug || it == RuntimeType.PartialEq || it == RuntimeType.Clone
         } + RuntimeType.Default
+    // Filter out attributes
+    private val builderAttributes =
+        metadata.additionalAttributes.filter {
+            it == Attribute.NonExhaustive
+        }
```

Comment:
```suggestion
    // Filter out any attribute that isn't `#[non_exhaustive]`.
    // `#[non_exhaustive]` is added to prevent users from instantiating the builder when it doesn't have any fields.
    private val builderAttributes =
        metadata.additionalAttributes.filter {
            it == Attribute.NonExhaustive
        }
```